### PR TITLE
[kubeadm/app/..add other packages]Switch to github.com/pkg/errors

### LINKED ISSUE #1183: Switch to github.com/pkg/errors
Currently kubeadm code base uses `fmt.Errorf` both for wrapping errors and for creating new errors, but there is a general consensus on moving toward `github.com/pkg/errors`

e.g.
```golang
// wrapping errors
// from
fmt.Errorf("couldn't create a temporary directory: %v", err)
// to
errors.Wrap(err, "couldn't create a temporary directory")

// creating new errors
// from
fmt.Errorf("couldn't initialize a Kubernetes cluster")
// to
errors. New("couldn't initialize a Kubernetes cluster")
```

@timothysc @rosti @neolit123 @chuckha @detiber @liztio 
If there is consensus, this is a good issue for new contributors that can do this package by packages (or grouping packages)

### PR DESCRIPTION
**Pull request by qingsenLi**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
> /kind cleanup

**What this PR does / why we need it**:
switch to github.com/pkg/errors the packages as follows:
cmd/kubeadm/app/constants
cmd/kubeadm/app/features
cmd/kubeadm/app/preflight

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes:kubernetes/kubeadm#1183

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by fabriziopandini** [comment]
@yuexiao-wang could you kindly confirm that this PR doesn't overlaps with your work/sync with @qingsenLi on this effort?
/hold

@qingsenLi please add Rif to the issue you are working on

**Comment by yuexiao-wang** [comment]
@fabriziopandini  i didn't modify these packages in my PR.

**Comment by yuexiao-wang** [comment]
@qingsenLi  you will need to run ./hack/update-gofmt.sh and ./hack/update-bazel.sh,  and amend your PR

**Comment by qingsenLi** [comment]
@fabriziopandini PTAL,thanks

File: cmd/kubeadm/app/constants/constants.go
Comment (yuexiao-wang) [inline cmd/kubeadm/app/constants/constants.go]: should be errors.Wrap

File: cmd/kubeadm/app/preflight/checks.go
Comment (yuexiao-wang) [inline cmd/kubeadm/app/preflight/checks.go]: errors.Wrap

File: cmd/kubeadm/app/preflight/checks.go
Comment (yuexiao-wang) [inline cmd/kubeadm/app/preflight/checks.go]: errors.Wrap

File: cmd/kubeadm/app/preflight/checks_test.go
Comment (yuexiao-wang) [inline cmd/kubeadm/app/preflight/checks_test.go]: don't move them

File: cmd/kubeadm/app/preflight/checks_test.go
Comment (yuexiao-wang) [inline cmd/kubeadm/app/preflight/checks_test.go]: packages from `github.com` are a independent group in `import`

File: cmd/kubeadm/app/preflight/checks_windows.go
Comment (yuexiao-wang) [inline cmd/kubeadm/app/preflight/checks_windows.go]: pkgerrors.Wrap

File: cmd/kubeadm/app/constants/constants.go
Comment (yuexiao-wang) [inline cmd/kubeadm/app/constants/constants.go]: errors.Wrap

**Review by yuexiao-wang — COMMENTED** [review]
@qingsenLi  thanks for your cleanup, and leave some comments

File: cmd/kubeadm/app/constants/constants.go
Comment (neolit123) [inline cmd/kubeadm/app/constants/constants.go]: kubernetes -> Kubernetes should be uppercase.

File: cmd/kubeadm/app/constants/constants_test.go
Comment (neolit123) [inline cmd/kubeadm/app/constants/constants_test.go]: kubernetes -> Kubernetes

File: cmd/kubeadm/app/features/features.go
Comment (neolit123) [inline cmd/kubeadm/app/features/features.go]: make Error -> error

File: cmd/kubeadm/app/preflight/checks_unix.go
Comment (neolit123) [inline cmd/kubeadm/app/preflight/checks_unix.go]: again, just change:
```
pkgerrors "github.com/pkg/errors"
```
-> 
```
"github.com/pkg/errors"
```

File: cmd/kubeadm/app/preflight/checks.go
Comment (neolit123) [inline cmd/kubeadm/app/preflight/checks.go]:
```
pkgerrors "github.com/pkg/errors"
```
->
```
"github.com/pkg/errors"
```

**Review by neolit123 — COMMENTED** [review]
@qingsenLi thanks, added some comments.

please also address the comments from @yuexiao-wang

File: cmd/kubeadm/app/preflight/checks_unix.go
Comment (qingsenLi) [reply cmd/kubeadm/app/preflight/checks_unix.go]: about this,add "github.com/pkg/errors" and then run make test WHAT=./cmd/kubeadm/app...
It shows errors:
cmd/kubeadm/app/preflight/checks_unix.go:22:2: imported and not used: "k8s.io/kubernetes/vendor/github.com/pkg/errors"
cmd/kubeadm/app/preflight/checks_unix.go:30:33: errors.New undefined (type []error has no field or method New)
so add pkgerrors "github.com/pkg/errors" instead of "github.com/pkg/errors".

**Review by fabriziopandini — APPROVED** [review]
@qingsenLi thanks for helping with this effort!
/approve

@neolit123 I leave the final lgtm to you; moreover, I think that it would be useful to make a quick code audit after this merge. wdyt?

File: cmd/kubeadm/app/preflight/checks_test.go
Comment (neolit123) [inline cmd/kubeadm/app/preflight/checks_test.go]: just `"github.com/pkg/errors"`

File: cmd/kubeadm/app/preflight/checks_unix.go
Comment (neolit123) [reply cmd/kubeadm/app/preflight/checks_unix.go]: https://github.com/pkg/errors/blob/master/errors.go#L102

this package does have a New function, something about the imports is not correct.
please try to fix it.

**Comment by neolit123** [comment]
@qingsenLi 

please try to fix `pkgerrors "github.com/pkg/errors"` -> `"github.com/pkg/errors"` in all files.
https://github.com/pkg/errors/blob/master/errors.go#L102

@fabriziopandini SGTM.

**Comment by yuexiao-wang** [comment]
@neolit123 if we don't use alias for `github.com/pkg/errors`,  it can't resolve for errors.Errorf when function has a named return as follows:
https://github.com/kubernetes/kubernetes/blob/02fe226fe4ef4b07d6da44579b508864be0059fa/cmd/kubeadm/app/preflight/checks_test.go#L176

**Comment by neolit123** [comment]
@yuexiao-wang 
> how do we deal with this case?

can we just rename such return variables - e.g.
`errors []error` -> `errorList []error`

**Comment by yuexiao-wang** [comment]
@neolit123 well， i agree with it. 
@qingsenLi  please try it

**Comment by qingsenLi** [comment]
@neolit123 please review again,thanks
I have fixed this PR.
pkgerrors "github.com/pkg/errors" -> "github.com/pkg/errors"
errors []error -> errorList []error

**Review by neolit123 — APPROVED** [review]
/lgtm
thank you for the update!

**Comment by neolit123** [comment]
@qingsenLi 
actually please squash the 2 commits into one first.

**Comment by neolit123** [comment]
/hold cancel
/lgtm

if the tests pass this will merge.
thanks for the updates @qingsenLi

**Pull request merged**
